lib/lightning/dashboard_metrics.ex

```diff
@@ -0,0 +1,161 @@
+defmodule Lightning.DashboardMetrics do
+  alias Lightning.Invocation.Run
+  alias Lightning.Repo
+  alias Lightning.Workflows.Workflow
+
+  alias Lightning.WorkOrder
+  import Ecto.Query
+
+  def get_metrics(project_id) do
+    # We could assume scheduled way to calculate these values
+    total_workorders = total_work_orders_in_last_30_days(project_id)
+    total_runs = aggregate_runs_for_project(project_id)
+    successful_runs = calculate_successful_runs_and_percentage(project_id)
+    failed_workorders = calculate_failed_work_orders_and_percentage(project_id)
+
+    [
+      total_workorders: total_workorders,
+      total_runs: total_runs,
+      successful_runs: successful_runs,
+      failed_workorders: failed_workorders
+    ]
+  end
+
+  def total_work_orders_in_last_30_days(project_id) do
+    thirty_days_ago = DateTime.utc_now() |> DateTime.add(-30 * 86400, :second)
+
+    count_query =
+      from(
+        wo in WorkOrder,
+        join: wf in Workflow,
+        on: wf.id == wo.workflow_id,
+        where: wf.project_id == ^project_id,
+        where: wo.inserted_at > ^thirty_days_ago,
+        select: count(wo.id)
+      )
+
+    Repo.one(count_query) || 0
+  end
+
+  def aggregate_runs_for_project(project_id) do
+    query =
+      from(
+        r in Run,
+        join: j in assoc(r, :job),
+        join: wf in assoc(j, :workflow),
+        where: wf.project_id == ^project_id,
+        group_by: r.exit_reason,
+        select: %{
+          exit_reason: r.exit_reason,
+          count: count(r.id)
+        }
+      )
+
+    runs = Repo.all(query)
+
+    {completed_runs, pending_running_runs} =
+      Enum.reduce(runs, {0, 0}, fn
+        %{exit_reason: "pending"} = run, {completed, pending_running} ->
+          {completed, pending_running + run.count}
+
+        %{exit_reason: "running"} = run, {completed, pending_running} ->
+          {completed, pending_running + run.count}
+
+        run, {completed, pending_running} ->
+          {completed + run.count, pending_running}
+      end)
+
+    %{completed_runs: completed_runs, pending_running_runs: pending_running_runs}
```

Comment:
You may be able to replace this with a filter expression, and have a count for each exit reason: 

https://hexdocs.pm/ecto/Ecto.Query.API.html#filter/2